libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridSplineToGrid.cpp

```diff
@@ -0,0 +1,904 @@
+                              curvilinearParameters.n_refinement, curvilinearParameters.m_refinement);
+    }
+
+    CurvilinearGrid grid(splines.m_projection);
+
+    // Make copy of spline because the may be changed
+    // i.e. the order may change and the number of spline points may increase.
+    Splines splinesCopy(splines);
+    VectorOfDoubleVectors splineIntersections(splines.GetNumSplines(), DoubleVector(splines.GetNumSplines(), 0.0));
+
+    UInt mRefinement = curvilinearParameters.m_refinement;
+    UInt nRefinement = curvilinearParameters.n_refinement;
+
+    // mn12 in Fortran.
+    VectorOfThreeInts splineInteraction(splines.GetNumSplines());
+    UInt numNSplines = 0;
+
+    ComputeSplineIntersections(splinesCopy, splineIntersections, numNSplines);
+    OrderSplines(splinesCopy, numNSplines, splineIntersections);
+    DetermineSplineOrientation(splinesCopy, numNSplines, splineIntersections, splineInteraction);
+    GenerateGrid(splinesCopy, splineIntersections, splineInteraction, numNSplines, mRefinement, nRefinement, grid);
+
+    return grid;
+}
+
+bool meshkernel::CurvilinearGridSplineToGrid::CheckSplines(const Splines& splines) const
+{
+    for (UInt splineIndex = 0; splineIndex < splines.GetNumSplines(); ++splineIndex)
+    {
+        if (splines.Size(splineIndex) < 2)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void meshkernel::CurvilinearGridSplineToGrid::DetermineIntersection(Splines& splines,
+                                                                    const UInt splineI,
+                                                                    const UInt splineJ,
+                                                                    UInt& numberTimesCrossing,
+                                                                    double& crossProductOfIntersection,
+                                                                    double& firstNormalisedIntersectionLength,
+                                                                    double& secondNormalisedIntersectionLength) const
+{
+    Point intersectionPoint;
+
+    // Need way of getting the number of times the spline intersect.
+    if (splines.GetSplinesIntersection(splineI, splineJ, crossProductOfIntersection, intersectionPoint, firstNormalisedIntersectionLength, secondNormalisedIntersectionLength))
+    {
+        numberTimesCrossing = 1;
+    }
+    else
+    {
+        secondNormalisedIntersectionLength = constants::missing::doubleValue;
+        firstNormalisedIntersectionLength = constants::missing::doubleValue;
+        crossProductOfIntersection = constants::missing::doubleValue;
+        numberTimesCrossing = 0;
+    }
+}
+
+void meshkernel::CurvilinearGridSplineToGrid::DoubleSplinePoints(Splines& splines) const
+{
+    std::vector<Point> doubledSplinePoints;
+    doubledSplinePoints.reserve(2 * splines.Size(splines.MaxSizeIndex()) - 1);
```

Comment:
2 looks like a factor that might deserve its variable, faction can be made more generic by not specifying the Double in the name of the function?

Reply:
This is possible, but for a more general increase factor, will considerably increase the complexity of this function.

Reply:
Changed, with comment on constant that the function IncreaseSplinePoints has to be changed to reflect the different scale factor.